### PR TITLE
Make allocation time use AbortController

### PR DESCRIPTION
**Pull request by ShogunPanda**
## What's in there

This PR streamlines heap sampling profiler and allocation time to both use AbortController.
The two tools now share most of the code so in the future we might also think to refactor them to avoid duplication.

Additionally, the `duration` option (which is now available even for allocation timeline) is ignored if a signal is provided.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -71,45 +71,31 @@ The available functions are:
...
+- `generateHeapSamplingProfile([options], [callback]): [Promise]`: Starts generating a heap sampling profiler. The valid options are:
 
   - `destination`: The path where to store the profile. The default will be a `.heapprofile` in the current directory.
   - `interval`: Sample interval, in bytes. Default is `32768` (32KB).
-  - `duration`: Sample duration, in milliseconds. Default is `10000` (10 seconds).
-  - `signal`: the [AbortController](http://npm.im/abort-controller) `signal`.
+  - `duration`: Sample duration, in milliseconds. Default is `10000` (10 seconds), and it is ignored if `signal` is provided.
+  - `signal`: The [AbortController](http://npm.im/abort-controller) `signal` to use to stop the operation.
+
+  The function accepts a callback function, otherwise it returns a Promise. The resolved value (or the callback argument) will be
```
Comment (salmanm) [inline README.md:81]: You have this on line 91 below. Did you forget to remove this one?

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@nearform/heap-profiler",
-  "version": "0.7.0",
+  "version": "0.8.0",
```
Comment (salmanm) [inline package.json:3]: Generally I've seen we don't update the version along with the change but not sure if that applies for this repo as well. Should we check with someone?

**Review by salmanm — APPROVED** [review]
I haven't looked at this module before so my understanding is limited for now. But this PR looks good to me.

File: README.md (hunk 1 above)
Comment (ShogunPanda) [reply README.md:81]: Nope. I know it's a little confusing, but these are two different tools which accept the almost the same arguments.
Thanks for noticing tho.

File: package.json (hunk 2 above)
Comment (ShogunPanda) [reply package.json:3]: I think we never established a convention. But probably you're right, we should not update this here. Reverting.

**Pull request merged**